CppSamples/EditData/EditGeodatabaseWithTransactions/EditGeodatabaseWithTransactions.cpp

```diff
@@ -0,0 +1,571 @@
+    {
+        if (featureType.name() == featureTypeName)
+        {
+            selectedFeatureType = featureType;
+            foundType = true;
+            break;
+        }
+    }
+
+    if (!foundType)
+    {
+        setMessageText(QString("Feature type '%1' not found in table '%2'.").arg(featureTypeName, tableName));
+        return;
+    }
+
+    // Convert screen point to map point
+    Point mapPoint = m_mapView->screenToLocation(location.x(), location.y());
+
+    if (mapPoint.isEmpty())
+    {
+        setMessageText("Invalid location clicked.");
+        return;
+    }
+
+    // Check if the point is within the geodatabase extent
+    if (m_extent)
+    {
+        // Convert point to WGS84 for comparison with extent (extent is in WGS84)
+        Point wgs84Point = mapPoint;
+        if (!mapPoint.spatialReference().isEmpty() && mapPoint.spatialReference().wkid() != 4326)
+        {
+            // Project the point to WGS84
+            Geometry projectedGeometry = GeometryEngine::project(mapPoint, SpatialReference::wgs84());
+            if (!projectedGeometry.isEmpty())
+            {
+                wgs84Point = static_cast<Point>(projectedGeometry);
+            }
+        }
+
+        bool withinBounds = (wgs84Point.x() >= m_extent->xMin() && wgs84Point.x() <= m_extent->xMax() &&
+                             wgs84Point.y() >= m_extent->yMin() && wgs84Point.y() <= m_extent->yMax());
+
+        if (!withinBounds)
+        {
+            setMessageText("Error: Feature geometry is outside the generate geodatabase geometry.");
+            return;
+        }
+    }
+
+    // Create feature using the selected type's template attributes
+    QVariantMap attributes;
+    if (!selectedFeatureType.templates().empty())
+    {
+        // Use the first template's attributes if available
+        attributes = selectedFeatureType.templates().at(0).prototypeAttributes();
+    }
+
+    Feature* newFeature = targetTable->createFeature(attributes, mapPoint);
+    if (!newFeature)
+    {
+        setMessageText("Failed to create new feature.");
+        return;
+    }
+
+    // Add the new feature to the table
+    QFuture<void> addFuture = targetTable->addFeatureAsync(newFeature);
+    QFutureWatcher<void>* addWatcher = new QFutureWatcher<void>(this);
+    connect(addWatcher, &QFutureWatcher<void>::finished, this, [this, addWatcher, targetTable]()
+    {
+        try
+        {
+            if (addWatcher && addWatcher->future().isValid() && !addWatcher->future().isCanceled())
+            {
+                setMessageText("Added feature.");
+
+                // Force refresh of the feature layer to show the new feature
+                if (m_mapView && m_mapView->map() && m_mapView->map()->operationalLayers())
+                {
+                    for (Layer* layer : *m_mapView->map()->operationalLayers())
+                    {
+                        if (FeatureLayer* featureLayer = qobject_cast<FeatureLayer*>(layer))
```

Comment:
prefer c++17 if+initializer syntax for these types of checks. The downside of this kind of check is scope bleed. 


```suggestion
                        if (FeatureLayer* featureLayer = qobject_cast<FeatureLayer*>(layer); featureLayer)
```